src/main/host/descriptor/socket/inet/mod.rs

```diff
@@ -429,10 +429,13 @@ impl InetSocketWeak {
 /// non-zero port will be chosen. The final local address will be returned. If the peer address is
 /// unspecified and has a port of 0, the socket will receive packets from every peer address. The
 /// socket will be automatically disassociated when the returned [`AssociationHandle`] is dropped.
+/// If `check_less_specific` is true, the association will also fail if there is already a socket
```

Comment:
Yeah I'm not super happy with this, but I don't really know a better way. I think ideally we'd get rid of the `is_interface_available` function altogether. I'm not sure what a better name is here, I think because this part of the code isn't really correct in the first place.

> I also wonder if there's a more descriptive name we could use for this parameter? Or if we could eliminate the bool parameter completely by having 2 different functions here (which might call an internal function with a bool parameter), or an enum?

We could make it two functions, but I don't know how to name those either. And these are functions that get passed through about 4 layers of things (inet module -> network namespace -> rust network interface -> C network interface) and I'm not sure where we'd make the 2 separate functions to 1 function with a bool distinction.

There's also a bigger problem where this part of the network interface is not working as intended anyway. This code in the network interface is supposed to prevent associating a wildcard (0.0.0.0) and non-wildcard peer address at the same time, but currently this check only works if you associate the wildcard address first, and then try associating the specific address. If you do it the other way around (associate a socket with the specific peer address and then associate a socket with a wildcard peer address), then the check doesn't work and you can incorrectly get two socket associations with the same local address when you shouldn't have.

So I don't want to make too many changes here, but if you think of a better name, or a nicer way to do this without restructuring too much of the code, let me know.